shared/src/main/scala/mlscript/syntax.scala

```diff
@@ -81,6 +81,7 @@ final case class Where(body: Term, where: Ls[Statement])             extends Ter
 final case class Forall(params: Ls[TypeVar], body: Term)             extends Term
 final case class Inst(body: Term)                                    extends Term
 final case class Super()                                             extends Term
+final case class Constructor(params: Tup, body: Ls[Statement])       extends Term // constructor(...) { ... }
```

Comment:
Why make this a Term instead of a statement?! Does something like `constructor(){} + 1` make sense to you?

Reply:
Use `final case class Constructor(params: Tup, body: Ls[Statement]) extends Statement // constructor(...) { ... }` instead.

Reply:
Okay, but the body should just be a `Term`, for simplicity and consistency.